openedx/features/edly/tests/test_middleware.py

```diff
@@ -0,0 +1,48 @@
+"""
+Unit tests for Middleware.
+"""
+from django.conf import settings
+from django.test import TestCase
+from django.test.client import Client, RequestFactory
+
+from openedx.features.edly import cookies
+from openedx.features.edly.tests.factories import EdlySubOrganizationFactory, EdlyUserFactory, SiteFactory
+
+
+class EdlyOrganizationAccessMiddlewareTests(TestCase):
+
+    def setUp(self):
+        self.user = EdlyUserFactory()
+        self.request = RequestFactory().get('/')
+        self.request.user = self.user
+        self.request.site = SiteFactory()
+
+        self.client = Client(SERVER_NAME=self.request.site.domain)
+        self.client.login(username=self.user.username, password='test')
+
+    def _create_edly_sub_organization(self):
+        """
+        Helper method to create 'EdlySubOrganization` for the request site.
+        """
+        return EdlySubOrganizationFactory(lms_site=self.request.site)
+
+    def test_user_with_organization_access(self):
+        """
+        Test edly user organization with access based on request.
+        """
+        self._create_edly_sub_organization()
+        self.client.cookies.load(
+            {
+                settings.EDLY_USER_INFO_COOKIE_NAME: cookies._get_edly_user_info_cookie_string(self.request)
+            }
+        )
+
+        response = self.client.get('/', follow=True)
+        assert response.status_code == 200
+
+    def test_user_without_organization_access(self):
+        """
+        Test edly user organization without access based on request.
```

Comment:
Test that logged in user gets 404 if user has no access for request site's edly sub organization.